plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployPreferencesPanel.java

```diff
@@ -105,14 +110,21 @@ public DeployPreferencesPanel(Composite parent, IProject project, Runnable layou
 
     Dialog.applyDialogFont(this);
 
-    GridDataFactory.fillDefaults().applyTo(this);
-    GridLayoutFactory.fillDefaults().generateLayout(this);
+    GridDataFactory.fillDefaults().grab(true, false).applyTo(this);
```

Comment:
Hmm, this shouldn't be here: a widget's Layout Data is set by whomever configures the composite that contains the widget.  It doesn't belong to the widget.  It can be placed in `DeployPropertyPage` and `DeployPreferencsDialog`.

(This can lead to `ClassCastExceptions` if it's at odds with the parent's layout.  For example, the parent may be using a `RowLayout`, not a `GridLayout`.)